tests/phpunit/api/v3/PaymentTest.php

```diff
@@ -406,4 +406,61 @@ public function testUpdatePayment() {
     ));
   }
 
+  /**
+   * Test create payment api for paylater contribution
+   */
+  public function testCreatePaymentPayLater() {
+    $contributionParams = array(
+      'total_amount' => 100,
+      'currency' => 'USD',
+      'contact_id' => $this->_individualId,
+      'financial_type_id' => 1,
+      'contribution_status_id' => 2,
+      'is_pay_later' => 1,
```

Comment:
We need an extra test to show that if the transaction is pending and a payment is added it transitions to 'Partially paid' 

I feel like the pay_later concept is kind of becoming obsolete and we should try to focus on what has been paid rather than the is_pay_later flag. Personally I think entirely removing that field should be a long-term goal in the way removing financial_type_id from the civicrm_contribution table should be